File: java/src/phase 1-conditional_thinking_java/Q22MiddleDigitRelation.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 3: Math and Number Logic
 * Question 22: Take a 3-digit number and determine if the middle digit is the largest, smallest, or neither.
 * 
 * Mentor Tips (Java 26):
 * - Extract three digits. Let middle be d2. Compare d2 against d1 (hundreds) and d3 (units).
 */
public class Q22MiddleDigitRelation {

    public static String solve(int num) {
        // TODO: Implement this method to solve the question
        return "";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("3-digit number: "); int num = scanner.nextInt();
                System.out.println("Middle relation: " + solve(num));
        scanner.close();
    }
}

File: python/phase 1-condtional_thinking/q22_middle_digit_relation.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 3: Math and Number Logic
Question 22: Take a 3-digit number and determine if the middle digit is the largest, smallest, or neither.

Mentor Tips (Python 3.14+):
- Extract three digits. Let middle be d2. Compare d2 against d1 (hundreds) and d3 (units).
"""

def solve(num: int) -> str:
    # TODO: Implement this method to solve the question
    return "neither"

def main():
    num = int(input("Enter a 3-digit number: "))\n    print(f"Middle relation: {solve(num)}")

if __name__ == "__main__":
    main()
